src/vmm/src/arch/aarch64/fdt.rs

```diff
@@ -70,6 +71,7 @@ pub fn create_fdt<T: DeviceInfoForFDT + Clone + Debug, S: std::hash::BuildHasher
     cmdline: CString,
     device_info: &HashMap<(DeviceType, String), T, S>,
     gic_device: &GICDevice,
+    vmgenid: &Option<VmGenId>,
```

Comment:
nothing to do with your PR, but I am quite fascinated by this `create_fdt` function
1. why is it parameterized by `BuildHasher`?
2. why does it return a `Vec<u8>` if it directly writes the data to guest memory? The only call-site of this function ignores the return value (although probably it should just return the `Vec<u8>` and have the callsite write it to guest memory. That way I imagine testing would also be easier)

Reply:
I agree. I can give a go at fixing it up. I'll resolve this for now though so I can merge the PR.

Reply:
Or maybe a good first issue?

Reply:
as well, yes. I'll do that.